src/items.rs

```diff
@@ -1262,6 +1262,29 @@ fn format_unit_struct(
     Some(format!("{}{};", header_str, generics_str))
 }
 
+fn set_brace_pos(has_fields: bool, has_content: bool, version: Version) -> BracePos {
+    return match version {
+        Version::One => {
+            if has_fields {
+                return BracePos::Auto;
+            }
+            BracePos::ForceSameLine
+        }
+        Version::Two => {
+            match (has_fields, has_content) {
+                // Doesn't have fields, there is nothing between {}, has generics.
+                (true, true) => BracePos::ForceSameLine,
+
+                // Doesn't have fields, has something between { }, has generics.
+                (true, false) => BracePos::Auto,
+
+                // Has fields, has generics.
+                (false, _) => BracePos::Auto,
+            }
+        }
+    };
+}
```

Comment:
Couple of notes:

1) the return in `return match version` is redundant, since the match is the only statement in this function. remove the trailing semicolon from the match statement to convert it to an expression and have it implicitly return its value.
2) I'm all for extracting this logic into a stand alone function, but I think we can simplify the logic see my next comment for details.